flink-ml-lib/src/test/java/org/apache/flink/ml/recommendation/SwingTest.java

```diff
@@ -0,0 +1,220 @@
+				Row.of("juice", " milk", 0, 1.3015095),
+				Row.of("juice", "orange", 0, 1.0717735),
+				Row.of("juice", "apple", 0, 1.0717735)
+			)
+		);
+
+	@Before
+	public void before() {
+		Configuration config = new Configuration();
+		config.set(ExecutionCheckpointingOptions.ENABLE_CHECKPOINTS_AFTER_TASKS_FINISH, true);
+		env = StreamExecutionEnvironment.getExecutionEnvironment(config);
+		env.getConfig().enableObjectReuse();
+		env.setParallelism(2);
+		env.enableCheckpointing(100);
+		env.setRestartStrategy(RestartStrategies.noRestart());
+		tEnv = StreamTableEnvironment.create(env);
+		Schema schema =
+			Schema.newBuilder()
+				.column("f0", DataTypes.STRING())
+				.column("f1", DataTypes.STRING())
+				.build();
+		DataStream <Row> dataStream = env.fromCollection(trainRows);
+		trainData = tEnv.fromDataStream(dataStream, schema).as("user", "item");
+		
+		Schema testSchema =
+			Schema.newBuilder()
+				.column("f0", DataTypes.STRING())
+				.build();
+		DataStream <Row> testStream = env.fromCollection(testRows);
+		testData = tEnv.fromDataStream(testStream, testSchema).as("item");
+	}
+
+	@Test
+	public void testParam() {
+		Swing swing = new Swing();
+
+		assertEquals("item", swing.getItemCol());
+		assertEquals("user", swing.getUserCol());
+		assertEquals(10, swing.getTopN());
+		assertEquals(10, swing.getMinUserItems());
+		assertEquals(1000, swing.getMaxUserItems());
+
+		swing.setItemCol("item_1")
+			.setUserCol("user_1")
+			.setTopN(20)
+			.setMinUserItems(10)
+			.setMaxUserItems(50);
+
+		assertEquals("item_1", swing.getItemCol());
+		assertEquals("user_1", swing.getUserCol());
+		assertEquals(20, swing.getTopN());
+		assertEquals(10, swing.getMinUserItems());
+		assertEquals(50, swing.getMaxUserItems());
+
+	}
+
+	@Test
+	public void testOutputSchema() {
+		Table input = trainData;
+		Swing swing = new Swing()
+			.setItemCol("item")
+			.setUserCol("user")
+			.setMinUserItems(1);
+		SwingModel swingModel = swing.fit(trainData);
+		Table output = swingModel.transform(input)[0];
+
+		assertEquals(
+			Arrays.asList("user","item", "SWING_ITEM_NAME", "SWING_ITEM_RANK", "SWING_ITEM_SCORE"),
+			output.getResolvedSchema().getColumnNames());
+	}
+
+	@Test
+	public void testFewerItemCase() {
+		Swing swing = new Swing()
+			.setItemCol("item")
+			.setUserCol("user")
+			.setMinUserItems(5);
+		SwingModel swingModel = swing.fit(trainData);
+		Table output = swingModel.setItemCol("item").transform(testData)[0];
+		List <Row> results = IteratorUtils.toList(output.execute().collect());
+		Assert.assertEquals(0, results.size());
+	}
+
+	@Test
+	public void testFitAndPredict() throws Exception {
+		Swing swing = new Swing()
+			.setItemCol("item")
+			.setUserCol("user")
+			.setMinUserItems(1);
+		SwingModel swingModel = swing.fit(trainData);
+		Table output = swingModel.setItemCol("item").transform(testData)[0];
+		List <Row> results = IteratorUtils.toList(output.execute().collect());
+		for(int i=0;i<results.size();i++){
```

Comment:
Maybe a better way to compare two `List<Row>` is to sort two lists with some deterministic comparator, then compare element pairs one by one.